code/modules/atmospherics/machinery/components/electrolyzer/electrolyzer_reactions.dm

```diff
@@ -89,30 +89,26 @@ GLOBAL_LIST_INIT(electrolyzer_reactions, electrolyzer_reactions_list())
 /datum/electrolyzer_reaction/halon_generation
 	name = "Halon generation"
 	id = "halon_generation"
-	desc = "Production of halon from CO2 and N2O"
+	desc = "Production of halon from the electrolysis of BZ."
 	requirements = list(
-		/datum/gas/carbon_dioxide = MINIMUM_MOLE_COUNT,
-		/datum/gas/nitrous_oxide = MINIMUM_MOLE_COUNT,
-		"MAX_TEMP" = 230
+		/datum/gas/bz = MINIMUM_MOLE_COUNT,
 	)
 	factor = list(
-		/datum/gas/carbon_dioxide = "2 moles of CO2 get consumed",
-		/datum/gas/nitrous_oxide = "1 mole of N2O gets consumed",
-		/datum/gas/halon = "1 mole of Halon gets produced",
-		"Energy" = "300 joules of energy is released per mole",
-		"Temperature" = "Can only occur under 230 kelvin.",
+		/datum/gas/bz = "Consumed during reaction.",
+		/datum/gas/oxygen = "0.2 moles of oxygen gets produced per mole of BZ consumed.",
+		/datum/gas/halon = "2 moles of Halon gets produced per mole of BZ consumed.",
+		"Energy" = "91.2321 kJ of thermal energy is released per mole of BZ consumed.",
+		"Temperature" = "Reaction efficiency is proportional to temperature.",
 		"Location" = "Can only happen on turfs with an active Electrolyzer.",
 	)
 
 /datum/electrolyzer_reaction/halon_generation/react(turf/location, datum/gas_mixture/air_mixture, working_power)
-
 	var/old_heat_capacity = air_mixture.heat_capacity()
-	air_mixture.assert_gases(/datum/gas/carbon_dioxide, /datum/gas/nitrous_oxide, /datum/gas/halon)
-	var/pressure = air_mixture.return_pressure()
-	var/reaction_efficency = min(1 / ((pressure / (0.5 * ONE_ATMOSPHERE)) * (max(air_mixture.gases[/datum/gas/carbon_dioxide][MOLES] / air_mixture.gases[/datum/gas/nitrous_oxide][MOLES], 1))), air_mixture.gases[/datum/gas/nitrous_oxide][MOLES], air_mixture.gases[/datum/gas/carbon_dioxide][MOLES] * INVERSE(2))
-	air_mixture.gases[/datum/gas/carbon_dioxide][MOLES] -= reaction_efficency * 2
-	air_mixture.gases[/datum/gas/nitrous_oxide][MOLES] -= reaction_efficency
-	air_mixture.gases[/datum/gas/halon][MOLES] += reaction_efficency
+	air_mixture.assert_gases(/datum/gas/bz, /datum/gas/oxygen, /datum/gas/halon)
+	var/reaction_efficency = min(air_mixture.gases[/datum/gas/bz][MOLES] * (1 - NUM_E ** (-0.5 * air_mixture.temperature * working_power / FIRE_MINIMUM_TEMPERATURE_TO_EXIST)), air_mixture.gases[/datum/gas/bz][MOLES])
```

Comment:
what is goin on here

Reply:
Faster the higher the temperature and electrolysis is, though it's decently fast in normal conditions anyways.